### DISCUSSION TITLE
How to generate JSON example like Swagger example tab?

### DISCUSSION THREAD

**Discussion by HamedFathi** [microsoft/OpenAPI.NET#893] [Q&A]
How can I convert an `OpenApiSchema` to a JSON example like what we have in Swagger?

![envlope](https://user-images.githubusercontent.com/8418700/172083342-abb0bcd8-394c-4825-b668-2681c29a6310.png)

![image](https://user-images.githubusercontent.com/8418700/172083371-15e614e4-92bc-4a79-9ae9-67d8600af403.png)

Is it possible to generate examples after reading any kind of (complex) schemas (types)? Any idea? any suggestion?

`OpenApiSchema.ToJsonExample()`

**Comment by darrelmiller** [comment]
We don't currently have this feature. Emitting just JSON would be fairly easy, but emitting good data into the schema is considerably harder.  I suspect the best approach would be to serialize the OpenAPISchema to JSON and then use a JSON Schema library that has already implemented the ability to emit examples.

**Reply by HamedFathi** [reply]
Is it possible to share an example with me about your best approach? (or at least Emitting just JSON with default values)
Honestly, I did something [here](https://github.com/HamedFathi/OpenApiExtended/blob/9078c7bba03c6c8bc263e60ab4f0fa355d918459/OpenApiExtended/OpenApiExtensions.cs#L2073) but I think it is so complicated! should be a better way!
